crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -159,6 +172,23 @@ impl Order {
     pub fn solver_determines_fee(&self) -> bool {
         self.is_partial() && matches!(self.kind, Kind::Limit { .. })
     }
+
+    /// The likelihood that this order will be fulfilled, based on token prices.
+    /// A larger value means that the order is more likely to be fulfilled.
+    /// This is used to prioritize orders when solving.
+    pub fn likelihood(&self, tokens: &auction::Tokens) -> num::BigRational {
+        match (
+            tokens.get(self.buy.token).price,
+            tokens.get(self.sell.token).price,
+        ) {
+            (Some(buy_price), Some(sell_price)) => {
+                let buy = buy_price.apply(self.buy.amount);
+                let sell = sell_price.apply(self.sell.amount);
+                conv::u256::to_big_rational(buy.0) / conv::u256::to_big_rational(sell.0)
```

Comment:
This could panic if `sell_amount` or `sell_price` is 0. **Should** never happen but it would be easy enough to account for that here.

Reply:
https://github.com/cowprotocol/services/pull/1678/commits/32c5a55cdf0695e5ef9293fee9f4206d14755443